lib/data/schema.js

```diff
@@ -641,9 +641,11 @@ const _updateEntityVersion = (xml, oldVersion, newVersion) => new Promise((pass,
 // If there are any problems with updating the XML, this will just
 // return the unaltered XML which will then be a clue for the worker
 // to not change anything about the Form.
+// 2022.1 -> 2024.1 forms only have version changed and suffix added.
+// 2023.1 -> 2024.1 forms (for updating) also have branchId and trunkVersion added.
 const updateEntityForm = (xml, oldVersion, newVersion, suffix) =>
   _updateEntityVersion(xml, oldVersion, newVersion)
-    .then(_addBranchIdAndTrunkVersion)
+    .then(x => (oldVersion === '2023.1.0' ? _addBranchIdAndTrunkVersion(x) : x))
     .then(x => addVersionSuffix(x, suffix))
     .catch(() => xml);
```

Comment:
Replace checking `2023.1` with a flag to add/not add branchId and trunkVersion so this function doesn't have to know about specific versions at all.